webapp/javascript/components/FlameGraph/FlameGraphComponent/Highlight.tsx

```diff
@@ -18,17 +18,26 @@ export interface HighlightProps {
   canvasRef: React.RefObject<HTMLCanvasElement>;
 }
 export default function Highlight(props: HighlightProps) {
-  const { canvasRef, barHeight, xyToHighlightData } = props;
+  const { canvasRef, barHeight, xyToHighlightData, zoom } = props;
   const [style, setStyle] = React.useState<React.CSSProperties>({
     height: '0px',
     visibility: 'hidden',
   });
 
+  React.useEffect(() => {
```

Comment:
Can you add a comment explaining a little bit more what's going on here?

From my understanding, after the `zoom` prop is set, we hide by default.
Then after moving any pixel it will trigger the eventListener, which will then highlight again.

Can you just put that into words? I think it makes it easier to reason about later.